linkerd/buffer/src/dispatch.rs

```diff
@@ -54,7 +54,7 @@ pub(crate) async fn run<S, Req, I>(
             e = idle().fuse() => {
                 let error = ServiceError(Arc::new(e.into()));
                 trace!(%error, "Idling out inner service");
-                return;
+                break;
```

Comment:
```suggestion
                return;
```

Reply:
way ahead of you, buddy https://github.com/linkerd/linkerd2-proxy/pull/759/commits/e2046e304e1beb7fb36911f3612765f9a597cc5e :D

Reply:
@hawkw i think that got the other break, but there were two.. so i think this is still relevant?